frontend/src/modules/dashboard/components/ValueInfo.tsx

```diff
@@ -240,13 +240,8 @@ const ControlValuesDisplay = ({
     alarmLimitsRequest === null
       ? undefined
       : ((alarmLimitsRequest as unknown) as Record<string, Range>)[stateKey];
-  const rangeValues = range === undefined ? { lower: '--', upper: '--' } : range;
-  const alarmTemp = alarmLimits === undefined ? { lower: '--', upper: '--' } : alarmLimits;
-  const alarmLimitsRange = alarmTemp as number[];
-  const { lower, upper } =
-    alarmLimitsRange?.length === 0
-      ? rangeValues
-      : { lower: alarmLimitsRange[0], upper: alarmLimitsRange[1] };
+  const rangeValues = range === undefined ? Range.fromJSON({ lower: '--', upper: '--' }) : range;
```

Comment:
I believe `Range`'s fields are supposed to be typed as `number`s, so `'--'` shouldn't be a valid value to provide for constructing a `Range`. I'm surprised the original code passed TypeScript. What would happen if you used `NaN` instead of `'--'` here?

Reply:
looks like `fromJson` takes `object: any` as parameter, so yes `NaN` also compiles.

Reply:
Any particular reason you chose to initialize with `0` rather than `NaN` in abe6334?

Reply:
Oh no reason as such! changed it to `NaN`